### PR TITLE
Add missing PathBase.

### LINKED ISSUE #1155: Api name not being catpured.
### Environment

How do you use Sentry?
Sentry SaaS (sentry.io) 

Which SDK and version?
.net core 3.8.3

### Steps to Reproduce

1. Did a post command to our wepapi.

### Expected Result
should have captured the api name as well as the path

https://ourservername.com/UserAPI/v1/Authentication/AuthenticateUser

### Actual Result

it only has references to the server and path, but not the api

   it only shows this:  v1/Authentication/AuthenticateUser

What actually happened. Maybe a screenshot/recording? Maybe some logs?

image

### PR DESCRIPTION
**Pull request by lucas-zimerman**
Refactor TryGetRouteTemplate and captured PathBase.

Close #1155.

### CODE REVIEW

**Review by josh-degraw — APPROVED** [review]
LGTM

File: src/Sentry.AspNetCore/Extensions/StringBuilderExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+using System.Text;
+
+namespace Sentry.AspNetCore.Extensions
+{
+    internal static class StringBuilderExtensions
+    {
+        internal static StringBuilder AppendIf(this StringBuilder builder, bool condition, string? value)
```
Comment (bruno-garcia) [inline src/Sentry.AspNetCore/Extensions/StringBuilderExtensions.cs:7]: You want to append null? If you have a condition already here doesn't it make sense to have `string value`?

File: test/Sentry.AspNetCore.Tests/Extensions/HttpContextExtensionsTests.cs [hunk 2]
```diff
@@ -8,14 +8,50 @@ namespace Sentry.AspNetCore.Tests.Extensions
...
+                if (pathBase is not null)
+                {
+                    // pathBase must start with '/' otherwise the new PathString will throw an exception.
+                    httpContext.Request.PathBase = new PathString(pathBase);
+                }
+                return httpContext;
+            }
+
+            public HttpContext GetMvcSut(string area = null, string controller = null, string action = null
+                , string pathBase = null)
```
Comment (bruno-garcia) [inline test/Sentry.AspNetCore.Tests/Extensions/HttpContextExtensionsTests.cs:25]: If they are too long for one line, please stack them up:

```suggestion
            public HttpContext GetMvcSut(
                string area = null,
                string controller = null,
                string action = null,
                string pathBase = null)
```

File: src/Sentry.AspNetCore/Extensions/HttpContextExtensions.cs [hunk 3]
```diff
@@ -39,9 +59,16 @@ internal static class HttpContextExtensions
...
             {
-                return !string.IsNullOrWhiteSpace(area)
-                    ? $"{area}.{controller}.{action}"
-                    : $"{controller}.{action}";
+                var hasArea = !string.IsNullOrWhiteSpace(area);
+                return new StringBuilder()
+                    .AppendIf(context.Request.PathBase.HasValue, context.Request.PathBase.Value?.TrimStart('/'))
+                    .AppendIf(context.Request.PathBase.HasValue, '.')
+                    .AppendIf(hasArea, area)
+                    .AppendIf(hasArea, '.')
```
Comment (bruno-garcia) [inline src/Sentry.AspNetCore/Extensions/HttpContextExtensions.cs:67]: Feels like overcomplicating things having this extension to what would be 2 `if` statements with two lines within each

File: test/Sentry.AspNetCore.Tests/Extensions/HttpContextExtensionsTests.cs [hunk 4]
```diff
@@ -21,7 +21,10 @@ public HttpContext GetSut(string pathBase = null)
                 return httpContext;
             }
 
-            public HttpContext GetMvcSut(string area = null, string controller = null, string action = null
+            public HttpContext GetMvcSut(
+                string area = null,
+                string controller = null,
+                string action = null
                 , string pathBase = null)
```
Comment (josh-degraw) [inline test/Sentry.AspNetCore.Tests/Extensions/HttpContextExtensionsTests.cs]:
```suggestion
                string action = null,
                string pathBase = null)
```

File: src/Sentry.AspNetCore/Extensions/HttpContextExtensions.cs [hunk 5]
```diff
@@ -37,11 +61,25 @@ internal static class HttpContextExtensions
...
-                    ? $"{area}.{controller}.{action}"
-                    : $"{controller}.{action}";
+                var builder = new StringBuilder();
+                if (context.Request.PathBase.HasValue)
+                {
+                    builder.Append(context.Request.PathBase.Value?.TrimStart('/'))
+                        .Append('.');
+                }
+
+                if (string.IsNullOrWhiteSpace(area) is false)
```
Comment (josh-degraw) [inline src/Sentry.AspNetCore/Extensions/HttpContextExtensions.cs:73]: Feels a bit verbose to do `is false` here
```suggestion
                if (!string.IsNullOrWhiteSpace(area))
```

**Review by josh-degraw — APPROVED** [review]
Just a few minor nits from me, otherwise lgtm

**Review by bruno-garcia — APPROVED** [review]

**Pull request merged**